Mutually exclusive validation incorrectly triggered when using baseline flag

**Describe the bug**
When using the dotnet-stryker command with the --with-baseline:<baseline> flag, it appears that the recently introduced validation within the SinceInput class ([Link](https://github.com/stryker-mutator/stryker-net/commit/47b373995213bdb1c56e04099939b033a9987b05#diff-28ba09420079cebe1f798288c0d525fb4e3316a3338949f06317ddd0259a401c)) is being triggered incorrectly. This issue seems to occur without explicitly using the since flag. This problem surfaced following the release of Stryker 3.11.0 and is resulting in a breaking change.

**Logs**
````
Version: 3.11.0

Stryker.NET failed to mutate your project. For more information see the logs 
below:


The since and baseline features are mutually exclusive.


##[error]PowerShell exited with code '1'.
````

**Expected behavior**
Not triggering this validation when using the baseline flag without the since flag.

**Version**
 - Stryker Version 3.11.0

--- a/src/Stryker.Core/Stryker.Core.UnitTest/Options/StrykerInputsTests.cs
+++ b/src/Stryker.Core/Stryker.Core.UnitTest/Options/StrykerInputsTests.cs
@@ -166,8 +166,8 @@ namespace Stryker.Core.UnitTest.Options
         [Fact]
         public void WithBaselineShouldNotThrow_2743() // https://github.com/stryker-mutator/stryker-net/issues/2743
         {
+            _target.ProjectVersionInput.SuppliedInput = "1";
             _target.WithBaselineInput.SuppliedInput = true;
-            _target.SinceInput.SuppliedInput = true;
 
             Should.NotThrow(() => _target.ValidateAll());
         }

--- a/src/Stryker.Core/Stryker.Core.UnitTest/Options/StrykerInputsTests.cs
+++ b/src/Stryker.Core/Stryker.Core.UnitTest/Options/StrykerInputsTests.cs
@@ -1,7 +1,7 @@
 using System.IO;
 using Shouldly;
-using Stryker.Core.Exceptions;
 using Stryker.Core.Baseline.Providers;
+using Stryker.Core.Exceptions;
 using Stryker.Core.Options;
 using Stryker.Core.Options.Inputs;
 using Xunit;
@@ -10,7 +10,7 @@ namespace Stryker.Core.UnitTest.Options
 {
     public class StrykerInputsTests : TestBase
     {
-        private StrykerInputs _target = new StrykerInputs()
+        private readonly StrykerInputs _target = new StrykerInputs()
         {
 
             AdditionalTimeoutInput = new AdditionalTimeoutInput(),
@@ -161,6 +161,15 @@ namespace Stryker.Core.UnitTest.Options
 
             var exception = Should.Throw<InputException>(() => _target.ValidateAll());
             exception.Message.ShouldBe("The since and baseline features are mutually exclusive.");
+        }
+
+        [Fact]
+        public void WithBaselineShouldNotThrow_2743() // https://github.com/stryker-mutator/stryker-net/issues/2743
+        {
+            _target.WithBaselineInput.SuppliedInput = true;
+            _target.SinceInput.SuppliedInput = true;
+
+            Should.NotThrow(() => _target.ValidateAll());
         }
 
         [Fact]

--- a/src/Stryker.Core/Stryker.Core/Options/Inputs/SinceInput.cs
+++ b/src/Stryker.Core/Stryker.Core/Options/Inputs/SinceInput.cs
@@ -12,7 +12,7 @@ namespace Stryker.Core.Options.Inputs
         {
             if (withBaseline.IsNotNullAndTrue())
             {
-                if (SuppliedInput.Value)
+                if (SuppliedInput.HasValue && SuppliedInput.Value)
                 {
                     throw new InputException("The since and baseline features are mutually exclusive.");
                 }

--- a/src/Stryker.Core/Stryker.Core/Options/Inputs/SinceInput.cs
+++ b/src/Stryker.Core/Stryker.Core/Options/Inputs/SinceInput.cs
@@ -12,7 +12,7 @@ namespace Stryker.Core.Options.Inputs
         {
             if (withBaseline.IsNotNullAndTrue())
             {
-                if (SuppliedInput.HasValue)
+                if (SuppliedInput.Value)
                 {
                     throw new InputException("The since and baseline features are mutually exclusive.");
                 }

